Tests/UnitTests/Purchasing/Purchases/PurchasesRestoreTests.swift

```diff
@@ -27,23 +27,26 @@ class PurchasesRestoreTests: BasePurchasesTests {
 
     func testRestoresDontPostMissingReceipts() {
         self.receiptFetcher.shouldReturnReceipt = false
-        var receivedError: NSError?
-        self.purchases.restorePurchases { (_, error) in
-            receivedError = error as NSError?
+
+        let receivedError = waitUntilValue { completed in
+            self.purchases.restorePurchases { (_, error) in
+                completed(error as NSError?)
+            }
         }
 
-        expect(receivedError?.code).toEventually(equal(ErrorCode.missingReceiptFileError.rawValue))
+        expect(receivedError).to(matchError(ErrorCode.missingReceiptFileError))
     }
 
     func testRestorePurchasesCallsCompletionOnMainThreadWhenMissingReceipts() {
         self.receiptFetcher.shouldReturnReceipt = false
-        var receivedError: NSError?
-        self.purchases.restorePurchases { (_, error) in
-            receivedError = error as NSError?
+        let receivedError = waitUntilValue { completed in
+            self.purchases.restorePurchases { (_, error) in
+                completed(error as NSError?)
+            }
         }
 
-        expect(self.mockOperationDispatcher.invokedDispatchOnMainThreadCount) == 1
-        expect(receivedError?.code).toEventually(equal(ErrorCode.missingReceiptFileError.rawValue))
+        expect(self.mockOperationDispatcher.invokedDispatchOnMainThreadCount) == 3
```

Comment:
why has this been changed to 3?

Reply:
The 1 was passing because it is 1 at some point. But in practice once the operation changes this became 3. I suppose I should change this to `>= 1` to match the same semantics as before, what do you think?

Reply:
Could `>= 1` give a false positive? Right now, there are 3 invokes on the main thread, one of them being the restoring one, which is the one we are testing for. If that one stops being called on the main thread (total of 2 invokes on main thread), the test will still pass

Reply:
Probably, but that was an issue with the existing test as well.

Reply:
I looked into this more. This just looks like a bad test. The same operation dispatcher is used for all of `Purchases`, so other background tasks are calling `dispatchOnMainThread` as well. The test has always passed because it was effectively checking `>= 1` with that `toEventually`. Eventually it was 1, *even before `restorePurchases` was called*, but that's not how the test was written.

~~Unfortunately can't think of a way to improve the test since we have no way to identify which operation was invoked.~~

Scratch that, of course there's a simple way haha

Reply:
Since the test is `testRestorePurchasesCallsCompletionOnMainThreadWhenMissingReceipts` I added this in the completion block:
```swift
expect(Thread.isMainThread) == true
``